### PR TITLE
Move new role for HSR trigger to main #10

### PR DESCRIPTION
**Pull request by rhmk**
New role for switching system replication

EDIT [SF]:
- PR also includes linting that removed EOL from many files
- PR also includes linting that removed whitespaces from many files

### CODE REVIEW

File: playbooks/sample-sap-fapolicy-all-update.yml [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-  
+
   tasks:
 
   # Update fapolicy for generic users
   - name: Fapolicy Update - generic
     vars:
       sap_fapolicy_type: "generic"
-    include_role: 
+    include_role:
```
Comment (rainerleber) [inline playbooks/sample-sap-fapolicy-all-update.yml:17]:
```suggestion
    ansible.builtin.include_role:
```
change to fqcn

File: playbooks/sample-sap-fapolicy-all-update.yml [hunk 2]
```diff
@@ -25,11 +25,11 @@
...
     register: sap_facts_register
 
-  # Update fapolicy for SAP users        
+  # Update fapolicy for SAP users
   - name: Fapolicy Update - SAP Users
     vars:
       sap_fapolicy_sid: "{{ item.Type }}"
       sap_fapolicy_type: "{{ item.Type }}"
-    include_role: 
+    include_role:
```
Comment (rainerleber) [inline playbooks/sample-sap-fapolicy-all-update.yml:33]:
```suggestion
    ansible.builtin.include_role:
```
change to fqcn

File: playbooks/sample-sap-firewall-all-update.yml [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
         param: "all"
     register: sap_facts_register
-        
+
   - name: Firewall Update
     vars:
       sap_firewall_type: "{{ item.Type }}"
       sap_firewall_instance_nr: "{{ item.InstanceNumber }}"
-    include_role: 
+    include_role:
```
Comment (rainerleber) [inline playbooks/sample-sap-firewall-all-update.yml:16]:
```suggestion
    ansible.builtin.include_role:
```
change to fqcn

File: playbooks/sample-sap-profile-update.yml [hunk 4]
```diff
@@ -27,11 +27,11 @@
...
     register: sap_facts_register
 
-  # SAP Profile Update      
+  # SAP Profile Update
   - name: SAP Profile Update
     vars:
       sap_update_profile_sid: "{{ item.SID }}"
       sap_update_profile_instance_nr: "{{ item.InstanceNumber }}"
-    include_role: 
+    include_role:
```
Comment (rainerleber) [inline playbooks/sample-sap-profile-update.yml:35]:
```suggestion
    ansible.builtin.include_role:
```

File: roles/sap_fapolicy/README.md [hunk 5]
```diff
@@ -40,20 +40,20 @@ Fapolicy entries will be updated to allow access to the following directories
...
-      
+
       tasks:
 
       # Update fapolicy for generic users
       - name: Fapolicy Update - generic
         vars:
           sap_fapolicy_type: "generic"
-        include_role: 
+        include_role:
```
Comment (rainerleber) [inline roles/sap_fapolicy/README.md:56]:
```suggestion
        ansible.builtin.include_role::
```

File: roles/sap_fapolicy/README.md [hunk 6]
```diff
@@ -66,20 +66,20 @@ Fapolicy entries will be updated to allow access to the following directories
...
         register: sap_facts_register
 
-      # Update fapolicy for SAP users        
+      # Update fapolicy for SAP users
       - name: Fapolicy Update - SAP Users
         vars:
           sap_fapolicy_sid: "{{ item.Type }}"
           sap_fapolicy_type: "{{ item.Type }}"
-        include_role: 
+        include_role:
```
Comment (rainerleber) [inline roles/sap_fapolicy/README.md:82]:
```suggestion
        ansible.builtin.include_role::
```

File: roles/sap_firewall/README.md [hunk 7]
```diff
@@ -48,19 +48,19 @@ Ansible role for updating firewall port entries based on SAP instance numbers
...
             param: "all"
         register: sap_facts_register
-            
+
       - name: Firewall Update
         vars:
           sap_firewall_type: "{{ item.Type }}"
           sap_firewall_instance_nr: "{{ item.InstanceNumber }}"
-        include_role: 
+        include_role:
```
Comment (rainerleber) [inline roles/sap_firewall/README.md:63]:
```suggestion
        ansible.builtin.include_role::
```

File: roles/sap_profile_update/README.md [hunk 8]
```diff
@@ -60,19 +60,19 @@ Ansible role for updating SAP profiles
...
           register: sap_facts_register
 
-        # Update all the profiles of the SAP systems in the host      
+        # Update all the profiles of the SAP systems in the host
         - name: Update all the profiles of the SAP systems in the host
           vars:
             sap_update_profile_sid: "{{ item.SID }}"
             sap_update_profile_instance_nr: "{{ item.InstanceNumber }}"
-          include_role: 
+          include_role:
```
Comment (rainerleber) [inline roles/sap_profile_update/README.md:75]:
```suggestion
          ansible.builtin.include_role::
```

**Review by rainerleber — COMMENTED** [review]
Could someone also have a look at the lint issues ?

File: roles/sap_hana_sr_takeover/.travis.yml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+  # Create ansible.cfg with correct roles_path
+  - printf '[defaults]\nroles_path=../' >ansible.cfg
+
+script:
+  # Basic role syntax check
+  - ansible-playbook tests/test.yml -i tests/inventory --syntax-check
+
+notifications:
+  webhooks: https://galaxy.ansible.com/api/v1/notifications/
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/.travis.yml:29]: do we need a travis.yml

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 10]
```diff
@@ -0,0 +1,112 @@
+---
+# tasks file for sap_hana_sr_takeover
+#
+
+- name: get hdb online state
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:5]:
```suggestion
- name: Get hdb online state
```

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+  shell: |
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:6]:
```suggestion
  ansible.builtin.shell: |
```

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+  args:
+    executable: /bin/bash
+  become: true
+  become_user: "{{ sap_hana_sid | lower }}adm"
+  changed_when: false
+  register: register_sap_hana_sr_takeover_online
+
+# returns: none, primary or replication method (on secondary)
+- name: get mode of system
+  shell: |
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:20]:
```suggestion
  ansible.builtin.shell: |
```

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+#
+- name: Fail, when planned primary cannot take over (no hsr configured)
+  fail:
+    msg: "This system is not part of HANA SR."
+  when:
+    - ansible_hostname == sap_hana_sr_takeover_primary
+    - register_sap_hana_sr_takeover_mode.stdout == "none"
+
+- name: make 'sap_hana_sr_takeover_primary' the new primary
+  shell: |
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:43]:
```suggestion
  ansible.builtin.shell: |
```

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+    - register_sap_hana_sr_takeover_mode.stdout != "primary"
+
+
+###
+# register the secondary
+#
+- name: register 'sap_hana_sr_takeover_secondary' to the new primary
+  block:
+    - name: Ensure HDB on 'sap_hana_sr_takeover_secondary' is stopped
+      shell: |
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:62]:
```suggestion
      ansible.builtin.shell: |
```

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+        source /usr/sap/{{ sap_hana_sid | upper }}/home/.sapenv.sh && \
+        /usr/sap/{{ sap_hana_sid | upper }}/HDB{{ sap_hana_instance_number }}/HDB stop
+      args:
+        executable: /bin/bash
+      when:
+        - register_sap_hana_sr_takeover_online.stdout == 'true'
+
+    - name: Register 'sap_hana_sr_takeover_secondary' to the primary
+      shell: |
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:71]:
```suggestion
      ansible.builtin.shell: |
```

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+
+  become: true
+  become_user: "{{ sap_hana_sid | lower }}adm"
+  when:
+    - ansible_hostname == sap_hana_sr_takeover_secondary
+    - register_sap_hana_sr_takeover_mode.stdout != sap_hana_sr_takeover_rep_mode
+
+## Check Online state and start HDB
+- name: get hdb online state
+  shell: |
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:91]:
```suggestion
  ansible.builtin.shell: |
```

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+    -sr_state | awk '($1 == "online:") {print $2}'
+  args:
+    executable: /bin/bash
+  become: true
+  become_user: "{{ sap_hana_sid | lower }}adm"
+  changed_when: false
+  register: register_sap_hana_sr_takeover_online
+
+- name: Ensure HDB is running
+  shell: |
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:104]:
```suggestion
  ansible.builtin.shell: |
```

File: roles/sap_hana_sr_takeover/tests/test-failover-hana1.yml [hunk 18]
```diff
@@ -0,0 +1,13 @@
+- name: Ensure hana1 is primary
+  hosts: hanas
+  become: true
+  tasks:
+    - name: Switch to hana1
+      include_role:
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tests/test-failover-hana1.yml:6]:
```suggestion
      ansible.builtin.include_role:
```

File: roles/sap_hana_sr_takeover/tests/test-failover-hana2.yml [hunk 19]
```diff
@@ -0,0 +1,13 @@
+- name: Ensure hana2 is primary
+  hosts: hanas
+  become: true
+  tasks:
+    - name: Switch to hana2
+      include_role:
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tests/test-failover-hana2.yml:6]:
```suggestion
      ansible.builtin.include_role:
```

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+  become: true
+  become_user: "{{ sap_hana_sid | lower }}adm"
+  changed_when: false
+  register: register_sap_hana_sr_takeover_mode
+
+##
+# enable the right primary
+#
+- name: Fail, when planned primary cannot take over (no hsr configured)
+  fail:
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:36]:
```suggestion
  ansible.builtin.fail:
```

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+        --replicationMode={{ sap_hana_sr_takeover_rep_mode }} \
+        --operationMode={{ sap_hana_sr_takeover_hsr_oper_mode }}
+      args:
+        executable: /bin/bash
+
+  become: true
+  become_user: "{{ sap_hana_sid | lower }}adm"
+  when:
+    - ansible_hostname == sap_hana_sr_takeover_secondary
+    - register_sap_hana_sr_takeover_mode.stdout != sap_hana_sr_takeover_rep_mode
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:87]:
```suggestion
```

File: roles/sap_hana_sr_takeover/tasks/main.yml [hunk 22]
```diff
@@ -0,0 +1,112 @@
...
+  when:
+    - ansible_hostname == sap_hana_sr_takeover_primary
+    - register_sap_hana_sr_takeover_mode.stdout != "primary"
+
+
+###
+# register the secondary
+#
+- name: register 'sap_hana_sr_takeover_secondary' to the new primary
+  block:
```
Comment (rainerleber) [inline roles/sap_hana_sr_takeover/tasks/main.yml:60]:
```suggestion
  become: true
  become_user: "{{ sap_hana_sid | lower }}adm"
  when:
    - ansible_hostname == sap_hana_sr_takeover_secondary
    - register_sap_hana_sr_takeover_mode.stdout != sap_hana_sr_takeover_rep_mode
  block:
```

File: roles/sap_hana_sr_takeover/tasks/main.yml (hunk 21 above)
Comment (rainerleber) [reply roles/sap_hana_sr_takeover/tasks/main.yml:87]: moved above the block definitions

Comment (rainerleber) [reply roles/sap_hana_sr_takeover/tasks/main.yml:87]: https://github.com/ansible/ansible-lint/blob/main/src/ansiblelint/rules/key_order.md

File: roles/sap_hana_sr_takeover/.travis.yml (hunk 9 above)
Comment (sean-freeman) [reply roles/sap_hana_sr_takeover/.travis.yml]: Agreed, Travis CI is not being used in the project at this time in favour of GH Actions - removed

**Review by sean-freeman — APPROVED** [review]
Reviewed existing comments and checked file diffs on existing files (mostly linting). Have not tested new Ansible Role, but code read-through suggests all good

**Pull request merged**